autonomic-plugin-common/src/test/java/br/com/autonomiccs/autonomic/plugin/common/services/AutonomiccsSystemVmDeploymentServiceTest.java

```diff
@@ -0,0 +1,1119 @@
+
+        Mockito.verify(spy).deploySystemVm(Mockito.anyLong(), Mockito.any(SystemVmType.class));
+        Mockito.verify(vmInstance).getManagementIpAddress();
+        Mockito.verify(sshUtils).executeCommandOnHostViaSsh(Mockito.eq("managementIp"), Mockito.eq("aptitude update"));
+        Mockito.verify(sshUtils).executeCommandOnHostViaSsh(Mockito.eq("managementIp"), Mockito.eq(commandToInstallOpenJDK7));
+        Assert.assertEquals(vmInstance, result);
+    }
+
+    @Test
+    public void deploySystemVmTest() {
+        HostVO host = Mockito.mock(HostVO.class);
+        VMTemplateVO systemVmTemplate = Mockito.mock(VMTemplateVO.class);
+        Account systemAcct = Mockito.mock(Account.class);
+        LinkedHashMap<Network, List<? extends NicProfile>> networks = new LinkedHashMap<>();
+        AutonomiccsSystemVm autonomiccsSystemVm = Mockito.mock(AutonomiccsSystemVm.class);
+
+        Mockito.when(host.getDataCenterId()).thenReturn(0l);
+        Mockito.when(host.getPodId()).thenReturn(0l);
+        Mockito.when(host.getClusterId()).thenReturn(0l);
+        Mockito.when(host.getId()).thenReturn(0l);
+
+        Mockito.doReturn(host).when(hostService).findHostById(Mockito.anyLong());
+        Mockito.doNothing().when(spy).validateParametersToDeployTheSystemVm(Mockito.anyLong(), Mockito.any(HostVO.class));
+        Mockito.doReturn(systemVmTemplate).when(spy).getSystemVmTemplate(Mockito.any(HostVO.class));
+        Mockito.doReturn(systemAcct).when(accountManager).getSystemAccount();
+        Mockito.doReturn(networks).when(spy).getSystemVmNetworks(Mockito.any(HostVO.class), Mockito.any(DataCenterDeployment.class), Mockito.any(Account.class));
+        Mockito.doReturn(0l).when(spy).getNextSystemVmId();
+        Mockito.doReturn("vmInstanceSuffix").when(spy).getVirtualMachineInstanceSuffix();
+        Mockito.doReturn("name").when(spy).createAutonomiccsSystemVmNameForType(Mockito.anyLong(), Mockito.any(SystemVmType.class), Mockito.anyString());
+        Mockito.doReturn(autonomiccsSystemVm).when(spy).createTheAutonomiccsSystemVm(Mockito.any(VMTemplateVO.class), Mockito.any(Account.class), Mockito.anyLong(),
+                Mockito.anyString());
+        Mockito.doReturn(autonomiccsSystemVm).when(spy).allocateAndStartTheSystemVm(Mockito.any(VMTemplateVO.class), Mockito.any(DataCenterDeployment.class), Mockito.eq(networks),
+                Mockito.any(AutonomiccsSystemVm.class));
+        Mockito.doNothing().when(spy).waitUntilTheAutonomiccsSystemVmIsUpAndRunning(Mockito.any(AutonomiccsSystemVm.class));
+
+        AutonomiccsSystemVm result = spy.deploySystemVm(0l, SystemVmType.ClusterManagerStartHostService);
+
+        InOrder inOrderHost = Mockito.inOrder(host);
+        inOrderHost.verify(host).getDataCenterId();
+        inOrderHost.verify(host).getPodId();
+        inOrderHost.verify(host).getClusterId();
+        inOrderHost.verify(host).getId();
+
+        InOrder inOrderSpy = Mockito.inOrder(spy);
```

Comment:
I believe the same point @rafaelweingartner made about using a single `InOrder` object in `br.com.autonomiccs.autonomic.plugin.common.guru.AutonomiccsSystemVmsGuruTest.configureBasicVmProfileTest()` can be applied here.